### ISSUE TITLE
Accessibility issues with Keybase website

### ISSUE THREAD

**Issue by ndarilek** [keybase/keybase-issues#1934]
I'm blind, and am attempting to use the Keybase website. I'd like to use your CLI but am having issues logging in with it, so in the meantime I'm attempting to resolve the issue from the website.

1. There are a number of unlabeled links (I.e. `<a/>` with no text content or something similar) which I'm guessing are icons. Please add `aria-label` atttributes to these links so my screen reader identifies them.
2. I'm attempting to edit/update my key by the web since I can't do so via the command line at the moment. The Edit link can receive keyboard focus but appears to pop up a list of additional links. Clicking on the link does nothing. This seems to be a mouseover interaction which is entirely inaccessible by the keyboard, since any action that takes focus away from this link causes the menu to disappear and become unreachable.

Thanks.

**Comment by zQueal** [comment]
Here's some [W3 documentation](http://www.w3.org/TR/wai-aria/states_and_properties) pertaining to `aria-label`. This should be pretty interesting!

As for the edit/update, I'm not entirely sure that it will end up working for you without being rewritten, unfortunately.

**Comment by malgorithms** [comment]
I'm working on issue 1, adding aria-labels to buttons without clear link text. Sorry!

> The Edit link can receive keyboard focus but appears to pop up a list of additional links. Clicking on the link does nothing. ... any action that takes focus away from this link causes the menu to disappear and become unreachable.

Interesting. I can fully get to these links and click them using just my keyboard, and no mouse. In my browser removing focus from the popup menu doesn't close it. Hmm! That makes it hard to debug. I'm using a bootstrap component for that popup.

**Comment by ndarilek** [comment]
Huh, OK, I take back #2, I am able to reach the links via keyboard. 
Earlier this morning I wasn't and the behavior was different. Had a pint 
of coffee between then and now so that's probably it. :)

Either way, might be worth marking that up with ARIA, maybe as a 
dropdown, then move keyboard focus to the first link when the Edit link 
is activated. Otherwise there is no indication that this is a dropdown. 
My expectation was that it'd pop up an entirely separate page, and I 
always do a bit of poking around the page when nothing happens to ensure 
that content wasn't appended/prepended to the DOM. See, for example, 
clicking on the key ID. Took a minute or so to find the text area 
containing the ASCII for the public key.

Thanks for the quick responses! This is encouraging.

**Comment by malgorithms** [comment]
can you check the aria-labels now?

**Comment by ndarilek** [comment]
Nice! Much better, thanks.

One link is still unlabeled. Looks like the link to edit pics near my 
full name.

**Comment by malgorithms** [comment]
ok, fixed the pic editing one. it should have the aria-label now too.

I haven't implemented a focus change yet as you said (that'll be a bit more work based on the way the contents of the dropdown are abstracted away), but as of 30 seconds ago I also added aria label to the dropdown links to suggest they're dropdowns. For example, the one for editing your PGP key. So that way it will be less confusing if you choose it (I hope).

**Comment by ndarilek** [comment]
Nice, thanks.

There are also labels missing from the device list page.

**Comment by smeveriss** [comment]
Hi.
The only issue I have is with the login/registration form.
I can click either form and they come up but if I try to switch forms because its all loaded within the website I can't easily get back to it.
As for the focus losing issue, I have noticed this on some forms but generally the site does work.
Now the client is another matter.
Single framed interface, with checkboxes not stating their status to nvda.
No indication when on different tabs they are not even being recognised as tabs.
As far as I can see the forms work as forms.
The interface is chromium based which helps I guess but there are a lot of unlabeled images all over the place.
The button for the windows system tray/notification area is not labeled at all.
Hitting space on that brings up a window which really is not accessible but you can use it to a point.
The biggest issue for me right now is when I move away from the client window then open keybase again.
The client loads and I can ocr it but as far as nvda states it can't be read.
I need to end all keybase tasks then reopen the client.
Of course maybe it would be easier if as an alternitive you could run keybase as a server at localhost like localhost:8888 which is the port for the resilio sync interface.
Thats working well enough like this or maybe make a more accessible client with all the extras.
The biggest issue seems that even when it does read the client is treated like a long word document or something.
There are no keyboard shortcuts though menus do work though I need to tab about these to find what I need.
I do wander if the client interface could just be a more powerfull version of the website interface or something.
Its worth noting, that the accessibility issues seem to only be windows ones as the guys that got me into keybase use smart phones and macs and have had no issue at all.
I am not sure about the chat, issue.
I am open to discussing and demoing this via a file, or via email to the devs.
I have seen a few issues reporting accessibility and there is a load of issues with the client.
In fact if I didn't have a use for it and if I couldn't access kbfs from explorer I wouldn't use it.
I use dropbox who's client is well not the best either and use that for my own personal and some collaboration shares.
Keybase is another backup to some other things namely for my podcasts.
What stops me actually using it completely for what its meant for is the client.
Its another tool for me to have for me.
But there are so many interface issues at least on windows its not like I will be recommending it to anyone unless you have a spaciffic purpose.
Right now for the blind files and sharing is probably the only reason that its even tolerated at all.
I do hope something is actually done, now zoom has actually tried and succeeded to improve its client.
An addon for nvda would be nice but I have another list to shove that on.

**Comment by alex19EP** [comment]
at the moment keybace has serious problems with the accessibility of android and electron applications. it would be great if someone worked on this.
